### PR TITLE
EVG-5432 Add weakly-typed mapstructure decoder

### PR DESCRIPTION
**Pull request by bsamek**

### CODE REVIEW

File: apimodels/agent_models.go [hunk 1]
```diff
@@ -93,7 +95,7 @@ type CreateHost struct {
 
 	// agent-controlled settings
 	CloudProvider       string `mapstructure:"provider" json:"provider" plugin:"expand"`
-	NumHosts            int    `mapstructure:"num_hosts" json:"num_hosts"`
+	NumHosts            string `mapstructure:"num_hosts" json:"num_hosts" plugin:"expand"`
```
Comment (tychoish) [inline apimodels/agent_models.go]: it feels super wrong to make/force callers (including from the rest v2 service to do this weird thing to our type system. 

Perhaps we should open a seperate ticket to make the expansion thing work with ints/

File: apimodels/agent_models.go [hunk 2]
```diff
@@ -129,10 +131,18 @@ func (ch *CreateHost) Validate() error {
 		catcher.Add(errors.New("aws_access_key_id, aws_secret_access_key, key_name must all be set or unset"))
 	}
 
-	if ch.NumHosts > 10 || ch.NumHosts < 0 {
-		catcher.Add(errors.New("num_hosts must be between 1 and 10"))
-	} else if ch.NumHosts == 0 {
-		ch.NumHosts = 1
+	if ch.NumHosts == "" {
+		ch.NumHosts = "1"
```
Comment (tychoish) [inline apimodels/agent_models.go]: this seems like it should be an error. It was before.

File: apimodels/agent_models.go (hunk 1 above)
Comment (bsamek) [reply apimodels/agent_models.go]: Done https://jira.mongodb.org/browse/EVG-5571.

File: apimodels/agent_models.go (hunk 2 above)
Comment (bsamek) [reply apimodels/agent_models.go]: That error check is preserved on 141.

**Pull request merged**